File: Assets/Greco3D/scripts/task/City.js
#pragma strict
import System.Collections.Generic;

class City
{
	public var stores = new List.<String>();
	public var coordsList = new List.<Coords>();
    public var trialList = new List.<Run>();
	var x =  new List.<float>();
    var y =  new List.<float>();

	public class Run{
		
		public var trials = new List.<int>();
		
		public function Run(tri:int[]){
			for(var i :int in tri){
			   this.trials.Add(tri[i]);
			}
		}
	}

	class Coords{
		var x =  new List.<float>();
    	var y =  new List.<float>();

		function Coords(X : List.<float>,Y : List.<float>){
		    this.x = X;
            this.y = Y;
		}
	}

    function City(name : String){
		BuildCity(name);
	}
	
    function BuildCity(cityName : String){

    var run1 : Run;
    var run2 : Run;
    var run3 : Run;
    var run4 : Run;

	var coords1 : Coords;
    var coords2 : Coords;
    var coords3 : Coords;
 	var coords4 : Coords;

	switch(cityName)
	{
	case 'CE':
		run1 = new Run([3,3,4,1,1,1,1,4,4,2,3,3,1,2,4,4,4,4,2,3,3,3,3,4]);
		run2 = new Run([3,2,2,1,4,4,2,3,4,4,4,4,1,1,2,1,3,3,1,2,2,1,3,3]);
		run3 = new Run([1,1,1,1,3,4,4,4,4,3,1,1,2,2,2,2,3,1,1,4,2,2,4,1]);
		run4 = new Run([1,1,1,4,3,3,3,3,2,4,3,3,2,2,2,2,1,3,2,2,2,2,4,4]);
		trialList = new List.<Run>([run1,run2,run3,run4]);
		
		//City 1
		x = new List.<float>([293.50,293.50,293.50,256.00,218.50,218.50,218.50,256.00]);
		y = new List.<float>([293.50,243.50,218.50,218.50,218.50,268.50,293.50,293.50]);
		coords1 = new Coords(x,y);
        
		//City 2
		x = new List.<float>([291.03,294.70,291.03,256.00,220.97,217.30,220.97,256.00]);
		y = new List.<float>([291.03,244.00,220.97,216.81,220.97,268.00,291.03,295.19]);
		coords2 = new Coords(x,y);

		//City 3
		x = new List.<float>([288.56,295.90,288.56,256.00,223.44,216.10,223.44,256.00]);
		y = new List.<float>([288.56,244.49,223.44,215.13,223.44,267.51,288.56,296.87]);
		coords3 = new Coords(x,y);

		//City 4
		x = new List.<float>([286.09,297.11,286.09,256.00,225.91,214.89,225.91,256.00]);
		y = new List.<float>([286.09,244.99,225.91,213.44,225.91,267.01,286.09,298.56]);
		coords4 = new Coords(x,y);

		coordsList = new List.<Coords>([coords1,coords2,coords3,coords4]);

		stores = new List.<String>(["Toy Store","1ST Bank", "Book Store", "Camera Store", "Coffee Shop","Craft Shop","Dentist","Fast Food Shop"]);
		break;
	case 'Greco':
		run1 = new Run([3,2,3,1,1,3,1,1,3,3,3,3,1,2,2,2,1,1,1,3,3,3,2,3,2]);
		run2 = new Run([1,2,1,2,1,3,3,1,1,1,2,2,2,3,2,3,3,1,2,2,1,1,3,1,1]);
		run3 = new Run([2,2,1,1,2,2,2,3,2,2,3,3,3,3,1,1,3,1,1,1,3,3,3,3,2]);
		run4 = new Run([1,1,2,2,2,3,3,3,2,1,1,3,3,2,1,2,3,1,3,3,1,1,1,1,3]);
		trialList = new List.<Run>([run1,run2,run3,run4]);

		x = new List.<float>([279.74,251.32,229.88,201.28,226.39,264.04,289.14,290.84,293.10,279.74]);
		y = new List.<float>([282.20,306.13,284.86,256.50,240.93,217.58,202.01,225.77,257.44,282.20]);
		coords1 = new Coords(x,y);

		x = new List.<float>([279.54,257.94,230.50,214.97,225.89,255.69,282.88,286.69,291.76,279.54]);
		y = new List.<float>([282.63,311.46,287.76,256.23,225.32,214.04,206.78,228.56,257.59,282.63]);
		coords2 = new Coords(x,y);
		
		x = new List.<float>([279.33,264.56,231.12,228.67,225.39,247.34,276.62,282.53,290.42,279.33]);
		y = new List.<float>([283.05,316.80,290.67,255.97,209.71,210.50,211.55,231.35,257.75,283.05]);
		coords3 = new Coords(x,y);

		coordsList = new List.<Coords>([coords1,coords2,coords3]);

		stores = new List.<String>(["Bakery","Cell Phones", "Gym", "Chinese Food", "Florist","Toy Store","Music Store", "Ice Cream", "Dentist"]);
		break;
	case 'Practice':

		run1 = new Run([1,1,2,1,1,2,1,2,2,1]);
		run2 = new Run([2,1,1,2,2,2,1,1,2,2]);
		trialList = new List.<Run>([run1,run2]);

		x = new List.<float>([267.54,234.22,257.60,270.85]);
		y = new List.<float>([273.49,274.33,227.98,241.26]);
		coords1 = new Coords(x,y);

		x = new List.<float>([272.77,234.36,248.31,269.79]);
		y = new List.<float>([276.62,263.94,235.01,242.62]);
		coords2 = new Coords(x,y);

		coordsList = new List.<Coords>([coords1,coords2]);
		
		stores = new List.<String>(["Book Store","Coffee Shop", "Craft Shop","1ST Bank"]);
		break;
	}
}
}

File: Assets/Standard Assets/City.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class City : MonoBehaviour {

    public List<string> stores = new List<string>();
    public List<Coords> coordsList = new List<Coords>();
    public List<Run> runList = new List<Run>();
    public List<float> x = new List<float>();
    public List<float> y = new List<float>();

    public City(string name)
    {
        BuildCity(name);
    }


    public class Run
    {
        public List<int> trials = new List<int>();

        public Run(List <int> tri)
        {
            trials = tri;
        }
    }

    public class Coords
    {
        public List<float> x = new List<float>();
        public List<float> y = new List<float>();

        public Coords(List<float> X, List<float> Y)
            {
                x = X;
                y = Y;
            }
    }

    void BuildCity(string name)
    {
        Run run1, run2, run3, run4;
        Coords coords1, coords2, coords3, coords4;

        switch (name)
        {
            case "CE":
                run1 = new Run(new List<int>() { 3, 3, 4, 1, 1, 1, 1, 4, 4, 2, 3, 3, 1, 2, 4, 4, 4, 4, 2, 3, 3, 3, 3, 4 });
                run2 = new Run(new List<int>() { 3, 2, 2, 1, 4, 4, 2, 3, 4, 4, 4, 4, 1, 1, 2, 1, 3, 3, 1, 2, 2, 1, 3, 3 });
                run3 = new Run(new List<int>() { 1, 1, 1, 1, 3, 4, 4, 4, 4, 3, 1, 1, 2, 2, 2, 2, 3, 1, 1, 4, 2, 2, 4, 1 });
                run4 = new Run(new List<int>() { 1, 1, 1, 4, 3, 3, 3, 3, 2, 4, 3, 3, 2, 2, 2, 2, 1, 3, 2, 2, 2, 2, 4, 4 });
                runList = new List<Run>() { run1, run2, run3, run4 };


                //City 1
                x = new List<float>() { 293.50f, 293.50f, 293.50f, 256.00f, 218.50f, 218.50f, 218.50f, 256.00f };
                y = new List<float>() { 293.50f, 243.50f, 218.50f, 218.50f, 218.50f, 268.50f, 293.50f, 293.50f };
                coords1 = new Coords(x, y);

                //City 2
                x = new List<float>() { 291.03f, 294.70f, 291.03f, 256.00f, 220.97f, 217.30f, 220.97f, 256.00f };
                y = new List<float>() { 291.03f, 244.00f, 220.97f, 216.81f, 220.97f, 268.00f, 291.03f, 295.19f };
                coords2 = new Coords(x, y);

                //City 3
                x = new List<float>() { 288.56f, 295.90f, 288.56f, 256.00f, 223.44f, 216.10f, 223.44f, 256.00f };
                y = new List<float>() { 288.56f, 244.49f, 223.44f, 215.13f, 223.44f, 267.51f, 288.56f, 296.87f };
                coords3 = new Coords(x, y);

                //City 4
                x = new List<float>() { 286.09f, 297.11f, 286.09f, 256.00f, 225.91f, 214.89f, 225.91f, 256.00f };
                y = new List<float>() { 286.09f, 244.99f, 225.91f, 213.44f, 225.91f, 267.01f, 286.09f, 298.56f };
                coords4 = new Coords(x, y);

                coordsList = new List<Coords>() { coords1, coords2, coords3, coords4 };
                stores = new List<string>() { "Toy Store", "1ST Bank", "Book Store", "Camera Store", "Coffee Shop", "Craft Shop", "Dentist", "Fast Food Shop" };
                break;
            case "Greco":
                run1 = new Run(new List<int>() { 3, 2, 3, 1, 1, 3, 1, 1, 3, 3, 3, 3, 1, 2, 2, 2, 1, 1, 1, 3, 3, 3, 2, 3, 2 });
                run2 = new Run(new List<int>() { 1, 2, 1, 2, 1, 3, 3, 1, 1, 1, 2, 2, 2, 3, 2, 3, 3, 1, 2, 2, 1, 1, 3, 1, 1 });
                run3 = new Run(new List<int>() { 2, 2, 1, 1, 2, 2, 2, 3, 2, 2, 3, 3, 3, 3, 1, 1, 3, 1, 1, 1, 3, 3, 3, 3, 2 });
                run4 = new Run(new List<int>() { 1, 1, 2, 2, 2, 3, 3, 3, 2, 1, 1, 3, 3, 2, 1, 2, 3, 1, 3, 3, 1, 1, 1, 1, 3 });
                runList = new List<Run>() { run1, run2, run3, run4 };

                x = new List<float>() { 279.74f, 251.32f, 229.88f, 201.28f, 226.39f, 264.04f, 289.14f, 290.84f, 293.10f, 279.74f };
                y = new List<float>() { 282.20f, 306.13f, 284.86f, 256.50f, 240.93f, 217.58f, 202.01f, 225.77f, 257.44f, 282.20f };
                coords1 = new Coords(x, y);

                x = new List<float>() { 279.54f, 257.94f, 230.50f, 214.97f, 225.89f, 255.69f, 282.88f, 286.69f, 291.76f, 279.54f };
                y = new List<float>() { 282.63f, 311.46f, 287.76f, 256.23f, 225.32f, 214.04f, 206.78f, 228.56f, 257.59f, 282.63f };
                coords2 = new Coords(x, y);

                x = new List<float>() { 279.33f, 264.56f, 231.12f, 228.67f, 225.39f, 247.34f, 276.62f, 282.53f, 290.42f, 279.33f };
                y = new List<float>() { 283.05f, 316.80f, 290.67f, 255.97f, 209.71f, 210.50f, 211.55f, 231.35f, 257.75f, 283.05f };
                coords3 = new Coords(x, y);

                coordsList = new List<Coords>() { coords1, coords2, coords3 };

                stores = new List<string> { "Bakery", "Cell Phones", "Gym", "Chinese Food", "Florist", "Toy Store", "Music Store", "Ice Cream", "Dentist" };
                break;

            case "Practice":
                print("setup practice");
                run1 = new Run(new List<int>() { 1, 1, 2, 1, 1, 2, 1, 2, 2, 1 });
                run2 = new Run(new List<int>() { 2, 1, 1, 2, 2, 2, 1, 1, 2, 2 });
                runList = new List<Run>() { run1, run2 };

                x = new List<float>() { 267.54f, 234.22f, 257.60f, 270.85f };
                y = new List<float>() { 273.49f, 274.33f, 227.98f, 241.26f };
                coords1 = new Coords(x, y);

                x = new List<float>() { 272.77f, 234.36f, 248.31f, 269.79f };
                y = new List<float>() { 276.62f, 263.94f, 235.01f, 242.62f };
                coords2 = new Coords(x, y);

                coordsList = new List<Coords>() { coords1, coords2 };

                stores = new List<string>() { "Book Store", "Coffee Shop", "Craft Shop", "1ST Bank" };
                break;
        }
        //Shuffle run order
        runList = Shuffle.ShuffleList(runList);
    }
}
